core/sail/inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/FastRdfsForwardChainingSailConnection.java

```diff
@@ -0,0 +1,1026 @@
+                    addInferredStatement(subject, RDFS.MEMBER, object);
+
+                    addInferredStatement(predicate, RDF.TYPE, RDFS.RESOURCE);
+                    addInferredStatement(predicate, RDF.TYPE, RDFS.CONTAINERMEMBERSHIPPROPERTY);
+                    addInferredStatement(predicate, RDF.TYPE, RDF.PROPERTY);
+                    addInferredStatement(predicate, RDFS.SUBPROPERTYOF, predicate);
+                    addInferredStatement(predicate, RDFS.SUBPROPERTYOF, RDFS.MEMBER);
+
+                }
+            } catch (NumberFormatException e) {
+                // Ignore exception.
+            }
+
+        }
+
+        if (actuallyAdd) {
+            connection.addStatement(subject, predicate, object, resources);
+
+        }
+
+        if (predicate.equals(RDF.TYPE)) {
+
+            resolveTypes((Resource) object)
+                .stream()
+                .peek(inferredType -> {
+                    if (fastRdfsForwardChainingSail.useAllRdfsRules && inferredType.equals(RDFS.CLASS)) {
+                        addInferredStatement(subject, RDFS.SUBCLASSOF, RDFS.RESOURCE);
+                    }
+
+                    inferRdfTypeSubject[0] = true;
+                })
+                .filter(inferredType -> !inferredType.equals(object))
+                .forEach(inferredType -> addInferredStatement(subject, RDF.TYPE, inferredType));
+        }
+
+        resolveProperties(predicate)
+            .stream()
+            .filter(inferredProperty -> !inferredProperty.equals(predicate))
+            .forEach(inferredProperty -> addInferredStatement(subject, inferredProperty, object));
+
+
+        if (object instanceof Resource) {
+            resolveRangeTypes(predicate)
+                .stream()
+                .peek(inferredType -> {
+                    if (fastRdfsForwardChainingSail.useAllRdfsRules && inferredType.equals(RDFS.CLASS)) {
+                        addInferredStatement(((Resource) object), RDFS.SUBCLASSOF, RDFS.RESOURCE);
+                    }
+                    inferRdfTypeObject[0] = true;
+
+                })
+                .forEach(inferredType -> addInferredStatement(((Resource) object), RDF.TYPE, inferredType));
+        }
+
+
+        resolveDomainTypes(predicate)
+            .stream()
+            .peek(inferredType -> {
+                if (fastRdfsForwardChainingSail.useAllRdfsRules && inferredType.equals(RDFS.CLASS)) {
+                    addInferredStatement(subject, RDFS.SUBCLASSOF, RDFS.RESOURCE);
+                }
+            })
+            .forEach(inferredType -> addInferredStatement((subject), RDF.TYPE, inferredType));
+
+    }
+
+
+    protected void addAxiomStatements() {
+        ValueFactory vf = fastRdfsForwardChainingSail.getValueFactory();
+
+        Statement statement = vf.createStatement(RDF.ALT, RDF.TYPE, RDFS.RESOURCE);
+        statementCollector(statement);
+        addInferredStatement(statement.getSubject(), statement.getPredicate(), statement.getObject());
```

Comment:
These are all the statements in RDFS forward chained. Since they are static, there is no need to do inferencing on the fly and it saves some time to just have them forward chained already.

I can add a comment stating that.